eyetracking/bids_scripts/et_prep_step1.py

```diff
@@ -0,0 +1,429 @@
+                    skip_run_num = True
+                    [sub, ses, fnum, task_type, appendix] = ev_file.split('_')
+                    run_num = run2task_mapping[task_root][task_type]
+                else:
+                    skip_run_num = False
+                    [
+                        sub,
+                        ses,
+                        fnum,
+                        task_type,
+                        run_num,
+                        appendix,
+                    ] = ev_file.split('_')
+
+                if sub in sub_list:
+                    assert sub == sub_num
+                    assert ses_num == ses
+
+                    log_list = glob.glob(
+                        f'{ses_path}/{sub_num}_{ses_num}_{fnum}.log'
+                    )
+                    has_log = len(log_list) == 1
+                    if has_log:
+                        with open(log_list[0]) as f:
+                            lines = f.readlines()
+                            empty_log = len(lines) == 0
+                    else:
+                        empty_log = True
+
+                    if skip_run_num:
+                        pupil_path = f'{ses_path}/{sub_num}_{ses_num}_{fnum}.pupil/{task_type}/000'
+                    else:
+                        pupil_path = f'{ses_path}/{sub_num}_{ses_num}_{fnum}.pupil/{task_type}_{run_num}/000'
+
+                    list_pupil = glob.glob(f'{pupil_path}/pupil.pldata')
+                    has_pupil = len(list_pupil) == 1
+                    if has_pupil:
+                        pupil_file_paths.append(
+                            (
+                                os.path.dirname(list_pupil[0]),
+                                (sub, ses, run_num, task_type, fnum),
+                            )
+                        )
+
+                    has_eyemv = len(glob.glob(f'{pupil_path}/eye0.mp4')) == 1
+                    has_gaze = len(glob.glob(f'{pupil_path}/gaze.pldata')) == 1
+
+                    run_data = [
+                        sub_num,
+                        ses_num,
+                        run_num,
+                        task_type,
+                        fnum,
+                        has_pupil,
+                        has_gaze,
+                        has_eyemv,
+                        has_log,
+                        empty_log,
+                    ]
+                    df_files = pd.concat(
+                        [
+                            df_files,
+                            pd.DataFrame(
+                                np.array(run_data).reshape(1, -1),
+                                columns=df_files.columns,
+                            ),
+                        ],
+                        ignore_index=True,
+                    )
+
+            except:
+                print(f'cannot process {ev_file}')
+
+    return df_files, pupil_file_paths
+
+
+def process_friends(
+    df_files: pd.DataFrame,
+    ses_list: list,
+) -> tuple:
```

Comment:
Why is there a separate/duplicate function for friends? Seems like a lot of the code is the same. What differs from the above generic case?